### DISCUSSION TITLE
GoCD Containers built for ARM64

### DISCUSSION THREAD

**Discussion by ThorstenEngelDeprecatedAccount** [gocd/gocd#10069] [Ideas]
Dear all,

I have to begin with a big thank you for GoCD. It was introduced in our company 4 or 5 years ago by a collegue how simply said: let's take this tool. Why? Because it is from ThoughtWorks. Okay, why is this so good? Because of Mike Fowler there, the collegue said. 

Well, it suits our needs and we are pretty happy with it. Use it even for triggering Kaniko-Jobs on K8s now with success in EKS.

What we really would like to have is a ThoughtWorks provided base image for ARM64 (server & alpine-agent). Do you have any pleans for adding those to the docker hub builds?

Warm regards from Munich,

thorsten

**Comment by arvindsv** [comment]
Hello. I would think this is unlikely. A recent comment which might be relevant is this: https://github.com/gocd/gocd/pull/9966#issuecomment-997408401

Also, GoCD is no longer officially connected to Thoughtworks. It's an independent open-source project.

**Reply by chadlwilson** [reply]
I agree with arvindsv in a general sense. From a pure technical curiosity perspective, I'd be interested in playing with this, but I don't have appropriate hardware to validate it on (except for paying for cloud infra).

I went down this mental path earlier and the blocker (or issues to resolve) were all the tools bundled with GoCD which then need to be built for aarch64 (and potentially packaged for the relevant distro in that format). Perforce client seems likely to be a problem.

Unless we can rationalise and drop distros and support for tools to replace with another the additional overhead is probably too much.

What would be the compelling reason to use aarch64 rather than x64 I wonder? (generally curious, I am not sure how folks are assessing the trade-offs with respect to general purpose software such as GoCD)

**Reply by ThorstenEngelDeprecatedAccount** [reply]
Thanks, I totally understand your reasoning. 

I probably will try to build the container myself and look if it will work ;-). Someone [did it](https://github.com/pellegrino/arm-gocd-agent) a few years ago. We use AWS and our agents run in EKS (managed kubernetes). As the ARM systems (graviton 2) really perform VERY WELL and have an attractive pricing, we try to get rid of x64 where possible.

As ARM is very energy-efficient I assume it will gain traction quickly in datacenters. The support for it is already very good (at least google seems to know even perforce clients for this architecture).

**Comment by ThorstenEngelDeprecatedAccount** [comment]
Thank you for providing release 23.1. Now I can directly use the centos-builds which are multi-arch and contain the arm64 builds.

**Reply by chadlwilson** [reply]
Good to hear! Became a lot easier to iterate on after I got an M1 MacBook for development which is largely how I was able to push through this 😀

**Reply by xlw712** [reply]
> Thank you for providing release 23.1. Now I can directly use the centos-builds which are multi-arch and contain the arm64 builds.

I also had a problem with the arm64. How did you solve it ？

**Reply by ThorstenEngelDeprecatedAccount** [reply]
Dear xlw712: My solution is pretty simple. My "clients" are build "FROM gocd/gocd-agent-debian-11:v23.1.0" and the server simply is "gocd/gocd-server-centos-9:v23.1.0". I am not using a MacBook, I am developing on Windows and my GoCD K8S-Cluster is running in AWS on ARM. If you need any further help, you're welcome to ask me.
